### ISSUE TITLE
MongoDB 3.x

### ISSUE THREAD

**Issue by Philmod** [kissjs/node-mongoskin#163]
Hey,

Is Mongoskin ready for MongoDB 3.x?

I've seen that it [requires](https://www.dropbox.com/s/o37ixmvgpm2jph3/Screenshot%202015-04-08%2012.44.58.png?dl=0) the 1.4.18 native module.

Thanks,
Philmod

**Comment by vkarpov15** [comment]
So mongoskin 1.4.13 should be compatible with any ~1.4 version of the mongodb npm module. Let me know if you run into any issues with that.

**Comment by nevf** [comment]
From what I can tell Mongoskin isn't compatible with MongoDB Native Driver V2 and hence MongoDB V3. For a start it fails in admin.js ```var Admin = require('mongodb').Admin;```. I haven't looked beyond this. 

IMHO it is easier and better to develop directly for the Native driver, especially seeing the number of pending issues with MongoSkin.

**Comment by vkarpov15** [comment]
Nope, latest mongodb driver ~1.4 also supports mongodb server 3.0.

**Comment by nevf** [comment]
To use Authentication with MongoDB V3 you need to use MongoDB Native Driver V2. ie. You can't access a Mongo V3 DB which uses authentication with the ~1.4 driver. My app uses authentication and therefore Mongoskin is a non-starter. FYI I am using agenda.js which uses Mongoskin and because of this issue agenda.js breaks. I'm currently refactoring agenda.js so it works directly with the MongoDB V2 driver.

**Comment by vkarpov15** [comment]
What authentication mechanism are you using and what sort of error are you getting? [Node driver >= 1.4.13 supports SCRAM-SHA-1](https://github.com/mongodb/node-mongodb-native/blob/master/HISTORY), so there should be no issue using auth with mongodb 3.0

**Comment by nevf** [comment]
When I passed authentication details in the url to agenda, which just passes them on to Mongoskin I got an error saying authentication failed and then agenda or mongoshell (not sure which) crashed with an exception. I am using the default SCRAM-SHA1 authentication and the new Wiredtiger DB engine.

**Comment by dhritzkiv** [comment]
I am also having issues ("auth failed") with the current `mongoskin` (through [`monq`](https://github.com/scttnlsn/monq)) when connecting to MongoDB 3.0 with SCRAM-SHA-1 authentication.

**Comment by nevf** [comment]
@dhritzkiv I think that mongoskin does auth against the db you are using with it. If instead you have the auth user specified in the admin db vs. the db you are using with mongoskin, then auth will fail. HTH.

**Comment by dhritzkiv** [comment]
@nevf Thanks for the tip. I'll check this out.

**Comment by jamesongithub** [comment]
any updates on supporting the 2.1.x driver?

**Comment by katmug** [comment]
I need to use MongoDB 3.X apis like insertMany, bulk operations etc. Are there any plans to support these features in mongoskin? if yes, by when can we expect it?

**Comment by thucnc** [comment]
Mongodb 3.6 even changed the open method, so I had to update the SkinCollection.prototype._open as following to adapt mongodb 3.6 driver.
```
this._skin_db.open(function(err, client) {
      var db = client.db(client.s.options.dbName)
      if(err) return callback(err);
      db.collection.apply(db, collection_args);
  });
```
Hope mongoskin officially support this change.

**Comment by jakehockey10** [comment]
Did anyone look into @thucnc 's fix?  That would be super cool if this worked and got merged.
